fireant/slicer/queries/special_cases.py

```diff
@@ -0,0 +1,77 @@
+from datetime import timedelta
+
+import pandas as pd
+
+from fireant.slicer.dimensions import DatetimeDimension
+from fireant.slicer.filters import RangeFilter
+from fireant.slicer.operations import RollingOperation
+
+
+def adjust_daterange_filter_for_rolling_window(dimensions, operations, filters):
```

Comment:
These functions could do with comments explaining why the date range has to be adjusted

Reply:
ok will do. Just hope I can find a better solution for this at some point because right now we need to jump through a lot of hoops to figure it out.